### ISSUE TITLE
Quota endpoint is returning '202 no content'`

### ISSUE THREAD

**Issue by gothub** [DataONEorg/bookkeeper#61]
The following request should return HTTP 404, but returns 202 instead:

```
curl -v -X GET "https://api.test.dataone.org:30443/bookkeeper/v1/quotas/10000" \
     -H "Origin: https://avatar.nceas.ucsb.edu" \
     -H "Authorization: Bearer $token"
...
< HTTP/1.1 204 No Content
< Server: nginx/1.19.0
< Date: Tue, 25 Aug 2020 21:29:13 GMT
< Connection: keep-alive
< Access-Control-Allow-Origin: https://avatar.nceas.ucsb.edu
< Access-Control-Allow-Methods: GET, POST, OPTIONS, PUT, DELETE
< Access-Control-Allow-Headers: Authorization, Content-Type, Origin, Cache-Control
< Access-Control-Allow-Credentials: true
```

If the quota is found, the proper HTTP result is returned.

**Comment by gothub** [comment]
In addition, for `QuotaResource.retrieve()` perform an authorization check.

**Comment by gothub** [comment]
An HTTP 404 status is returned if the requested quota is not found. Fixed in commit 9320122ea85b7ef4e9e8fc1e20dea0b05b0b12d8
